File: LeetCode/Easy/0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        table = {}

        for i in range(0, len(nums)):
            diff = target - nums[i]

            if diff in table:
                return [table[diff], i]

            table[nums[i]] = i

        return []

File: LeetCode/Easy/0001-two-sum/0001-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> table;

        for (int i = 0; i < nums.size(); i++) {
            int diff = target - nums[i];

            if (table.contains(diff)) {
                return { table[diff], i };
            }

            table[nums[i]] = i;
        }

        return {};
    }
};